eng/pipelines/templates/steps/compilation.check.yml

```diff
@@ -0,0 +1,31 @@
+parameters:
+  - name: Artifact
+    type: object
+    default: {}
+  - name: ArtifactName
+    type: string
+    default: "not-specified"
+  - name: RetryLimit
+    type: number
+    default: 30
+
+steps:
+  - download: current
+    artifact: ${{parameters.ArtifactName}}-signed
+  - pwsh: |
+      $pomf = Get-ChildItem "$(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{parameters.Artifact.groupId}}/${{parameters.Artifact.name}}/*.pom"
```

Comment:
Since this directory path is so long and used in a couple places, it might make sense to define it as a variable or parameter instead.

Reply:
Also we should add some validation to ensure we get back one and only one file that we expect.

Reply:
mvn compile will exit with error, but i guess stopping early on and not run for 30 min is better